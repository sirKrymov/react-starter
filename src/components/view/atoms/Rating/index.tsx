// @SETUP: Please don't forget to remove this component or current comment if component is used in project
// @TODO: Add components tests
import React, { ReactElement } from 'react';

import { IoIosStarOutline, IoIosStar } from 'react-icons/io';
import RatingComponent from 'react-rating';

import { ReactIcon } from 'components/view/atoms/ReactIcon';

import './styles.scss';

interface IProps {
  onChange(number: number): void;
  value: number;
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl' | 'xxxl';
}

export function Rating({ onChange, value, size }: IProps): ReactElement {
  return (
    <div className="rating">
      <RatingComponent
        initialRating={value}
        emptySymbol={
          <ReactIcon size={size}>
            <IoIosStarOutline />
          </ReactIcon>
        }
        fullSymbol={
          <ReactIcon>
            <IoIosStar size={size} />
          </ReactIcon>
        }
        onChange={onChange}
      />
    </div>
  );
}
