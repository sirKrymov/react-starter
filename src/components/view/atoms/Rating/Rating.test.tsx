import React from 'react';

import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { Rating } from '../Rating';

describe('<Rating> compound component', () => {
  const handleChange = jest.fn();

  let component: ReactTestRenderer;
  let shallowComponent: ShallowWrapper;

  const createComponent = () => {
    component = renderer.create(<Rating onChange={handleChange} value={0} />);
  };

  const createShallowComponent = () => {
    shallowComponent = shallow(<Rating onChange={handleChange} value={4} />);
  };

  describe('WHEN the component loads', () => {
    it('THEN it should render without crashing', () => {
      createComponent();
      expect(component.toJSON()).toBeDefined();
    });

    it('THEN it should match snapshot', () => {
      createShallowComponent();
      expect(shallowToJson(shallowComponent)).toMatchSnapshot();
    });
  });
});
