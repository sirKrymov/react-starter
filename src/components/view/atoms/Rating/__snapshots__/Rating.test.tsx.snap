// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating> compound component WHEN the component loads THEN it should match snapshot 1`] = `
<div
  className="rating"
>
  <RatingAPILayer
    direction="ltr"
    emptySymbol={
      <ReactIcon>
        <IoIosStarOutline />
      </ReactIcon>
    }
    fractions={1}
    fullSymbol={
      <ReactIcon>
        <IoIosStar />
      </ReactIcon>
    }
    initialRating={4}
    onChange={[MockFunction]}
    onClick={[Function]}
    onHover={[Function]}
    placeholderSymbol={
      Object {
        "backgroundColor": "red",
        "border": "5px double white",
        "borderRadius": "50%",
        "display": "inline-block",
        "height": 30,
        "width": 30,
      }
    }
    quiet={false}
    readonly={false}
    start={0}
    step={1}
    stop={5}
  />
</div>
`;
